### PR TITLE
Update 2020-04-01-preview swagger and add additional example files

### PR DESCRIPTION
**Pull request by ahamad-MS**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [X] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [X] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by sanjaiganesh** [comment]
>           {

consider moving common parameters under "path", instead of defining on every operation

---
Refers to: specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-04-01-preview/EventGrid.json:797 in 8b16c5c. [](commit_id = 8b16c5c50d175f5eefff20ca98e743d0bcf6624d, deletion_comment = False)

**Comment by sanjaiganesh** [comment]
>           "201": {

is this operation async  ? 
1.If so, considering always using async for this. easy for clients to write code to target your apis.
2. annotate
"x-ms-long-running-operation": true,
        "x-ms-long-running-operation-options": {
          "final-state-via": "azure-async-operation"  
        }

---
Refers to: specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-04-01-preview/EventGrid.json:817 in 8b16c5c. [](commit_id = 8b16c5c50d175f5eefff20ca98e743d0bcf6624d, deletion_comment = False)

File: specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-04-01-preview/EventGrid.json [hunk 1]
```diff
@@ -695,6 +698,260 @@
...
+            "description": "*** Error Responses: ***\n\n * 400 Bad Request.\n\n * 500 Internal Server Error."
+          }
+        },
+        "x-ms-examples": {
+          "EventChannels_CreateOrUpdate": {
+            "$ref": "./examples/EventChannels_CreateOrUpdate.json"
+          }
+        }
+      },
+      "delete": {
```
Comment (sanjaiganesh) [inline specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-04-01-preview/EventGrid.json]: is this a tracked resource ? (ARM routing type)
If so, it must support PATCH (at least tags)

**Comment by sanjaiganesh** [comment]
>         "operationId": "SystemTopicEventSubscriptions__ListBySystemTopic",

double _

---
Refers to: specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-04-01-preview/EventGrid.json:1602 in 8b16c5c. [](commit_id = 8b16c5c50d175f5eefff20ca98e743d0bcf6624d, deletion_comment = False)

**Comment by sanjaiganesh** [comment]
>         "operationId": "PartnerTopicEventSubscriptions__ListByPartnerTopic",

double _

---
Refers to: specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-04-01-preview/EventGrid.json:1971 in 8b16c5c. [](commit_id = 8b16c5c50d175f5eefff20ca98e743d0bcf6624d, deletion_comment = False)

**Comment by sanjaiganesh** [comment]
>       "post": {

is paging required ? if you anticipate this, add x-ms-paging

---
Refers to: specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-04-01-preview/EventGrid.json:2992 in 8b16c5c. [](commit_id = 8b16c5c50d175f5eefff20ca98e743d0bcf6624d, deletion_comment = False)

**Comment by sanjaiganesh** [comment]
>         ],

Get /skus ARM has the api .. checkout armwiki for the contract [https://armwiki.azurewebsites.net/api_contracts/SKUUpdateContract%E2%80%8B%E2%80%8B.html?q=sku](https://armwiki.azurewebsites.net/api_contracts/SKUUpdateContract%E2%80%8B%E2%80%8B.html?q=sku)

---
Refers to: specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-04-01-preview/EventGrid.json:4845 in 8b16c5c. [](commit_id = 8b16c5c50d175f5eefff20ca98e743d0bcf6624d, deletion_comment = False)

**Comment by sanjaiganesh** [comment]
>         }

mark all the secrets with x-ms-secret
and also ensure they are not present in get response

consider x-ms-mutability : write, update

---
Refers to: specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-04-01-preview/EventGrid.json:7677 in 8b16c5c. [](commit_id = 8b16c5c50d175f5eefff20ca98e743d0bcf6624d, deletion_comment = False)

**Comment by sanjaiganesh** [comment]
> {

pls consider my feedback for allt he apis. there are lot of updates. i didn't examine each and every api.

Ensure listBySub and listByRg are present wherever applicable..

---
Refers to: specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-04-01-preview/EventGrid.json:1 in 8b16c5c. [](commit_id = 8b16c5c50d175f5eefff20ca98e743d0bcf6624d, deletion_comment = False)

**Comment by ahamad-MS** [comment]
> "201": {
> 
> 
> is this operation async  ?
> 1.If so, considering always using async for this. easy for clients to write code to target your apis.
> 2. annotate
> "x-ms-long-running-operation": true,
> "x-ms-long-running-operation-options": {
> "final-state-via": "azure-async-operation"   or "location" based on the header used to track long running operation
> }
> 
> Refers to: specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-04-01-preview/EventGrid.json:817 in 8b16c5c. [](commit_id = 8b16c5c, deletion_comment = False)


==> This is planned to be async eventually before GA. anyhow, I am removing 201 for the time being. thx

**Comment by ahamad-MS** [comment]
> {
> 
> 
> consider moving common parameters under "path", instead of defining on every operation
> 
> Refers to: specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-04-01-preview/EventGrid.json:797 in 8b16c5c. [](commit_id = 8b16c5c, deletion_comment = False)

===> We are getting these auto-generated using auto-rest and all previous API versions using same style. Thanks for the feedback but we will keep this as generated.

File: specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-04-01-preview/EventGrid.json (hunk 1 above)
Comment (ahamad-MS) [reply specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-04-01-preview/EventGrid.json]: No, event channels is a nested resource under partnerNamespaces. its type is:
"Microsoft.EventGrid/partnerNamespaces/eventChannels"

**Comment by ahamad-MS** [comment]
> "operationId": "SystemTopicEventSubscriptions__ListBySystemTopic",
> 
> 
> double _
> 
> Refers to: specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-04-01-preview/EventGrid.json:1602 in 8b16c5c. [](commit_id = 8b16c5c, deletion_comment = False)

===> Thanks.. fixed in second iteration/commit

**Comment by ahamad-MS** [comment]
> "operationId": "PartnerTopicEventSubscriptions__ListByPartnerTopic",
> 
> 
> double _
> 
> Refers to: specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-04-01-preview/EventGrid.json:1971 in 8b16c5c. [](commit_id = 8b16c5c, deletion_comment = False)


===> Thanks.. fixed in second iteration/commit

**Comment by ahamad-MS** [comment]
> "post": {
> 
> 
> is paging required ? if you anticipate this, add x-ms-paging
> 
> Refers to: specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-04-01-preview/EventGrid.json:2992 in 8b16c5c. [](commit_id = 8b16c5c, deletion_comment = False)


===> No, pagination is not supported for PartnerNamespaces_ListSharedAccessKeys as this returns max of two values: key1, and key1. Same pattern is used with other existing resources: domains and topics.

**Comment by ahamad-MS** [comment]
> ],
> 
> 
> Get /skus ARM has the api .. checkout armwiki for the contract https://armwiki.azurewebsites.net/api_contracts/SKUUpdateContract%E2%80%8B%E2%80%8B.html?q=sku
> 
> Refers to: specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-04-01-preview/EventGrid.json:4845 in 8b16c5c. [](commit_id = 8b16c5c, deletion_comment = False)


===> yes, we are aware of this operation and our service has support for it. however, we are under the impression that this is used by ARM only and hence we are not exposing it to regular customers. If this is not the case, we will consider adding this in future API changes but not in this one. thanks

**Comment by ahamad-MS** [comment]
> x-ms-secret


===> We are modeling this similar to existing ListKey and regenerate key operations for existing resources: domains and topics. We need to return these keys to users hence not sure if this is applicable to our case here. Won't fix.

**Comment by ahamad-MS** [comment]
> {
> 
> pls consider my feedback for allt he apis. there are lot of updates. i didn't examine each and every api.
> Ensure listBySub and listByRg are present wherever applicable..
> 
> Refers to: specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-04-01-preview/EventGrid.json:1 in 8b16c5c. [](commit_id = 8b16c5c, deletion_comment = False)


===> thanks for the review and feedback. Unfortunately, swagger reorder entries randomly and we ended up with existing resources being replaced by new resources hence the diff is big. Most of the changes are already existing from previous API versions (e.g., event subscriptions, domains, domain topics, topics, etc).

**Comment by ahamad-MS** [comment]
@fengzhou-msft .. Can you please help merge/complete this PR. We address most of the valid errors and remaining ones are either warnings for existing properties or things we can take care of in next PR. We need to catch some hard deadlines and would like your help here. thanks

**Comment by fengzhou-msft** [comment]
@ahamad-MS there's a required CI check failed, please check the [error](https://dev.azure.com/azure-sdk/public/_build/results?buildId=283191&view=logs&jobId=688669d0-441c-57c3-cf6d-f89a22ccfa5d&j=688669d0-441c-57c3-cf6d-f89a22ccfa5d&t=923b5a90-e462-5994-c166-86509c6bf5f2) and fix it, then I can go on to merge the PR.

**Comment by ahamad-MS** [comment]
@fengzhou-msft all required tests are passing,. kindly provide final sign off and merge. thanks

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**